### PR TITLE
add do_where for PDB, update whatsnew tool.

### PR DESCRIPTION
**Pull request by Carreau**

### CODE REVIEW

File: tools/update_whatsnew.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 import io
 import os
+from glob import glob
```
Comment (Carreau) [inline tools/update_whatsnew.py]: I use glob, otherwise listdir picks up my `*.swp~` files.

File: tools/update_whatsnew.py [hunk 2]
```diff
@@ -20,15 +21,15 @@
 
 # 1. Collect the whatsnew snippet files ---------------------------------------
 
-files = set(os.listdir(pr_dir))
+files = set()
+for f in glob(pjoin(pr_dir, '*.rst')):
+    files.add(f)
```
Comment (takluyver) [inline tools/update_whatsnew.py]: Why the for loop? Can't it just be:

```python
files = set(glob(pjoin(pr_dir, '*.rst')))
```

File: tools/update_whatsnew.py [hunk 3]
```diff
@@ -20,15 +21,15 @@
...
 # Ignore explanatory and example files
-files.difference_update({'README.md',
+files.difference_update({pjoin(pr_dir, f) for f in {'README.md',
```
Comment (takluyver) [inline tools/update_whatsnew.py]: No need to subtract `README.md` now it's only finding .rst files.

File: tools/update_whatsnew.py [hunk 4]
```diff
@@ -38,7 +39,11 @@ def getmtime(f):
 features, incompats = [], []
 for path in files:
     with io.open(path, encoding='utf-8') as f:
-        content = f.read().rstrip()
+        try:
+            content = f.read().rstrip()
+        except Exception as e:
+            raise Exception('Error reading "{}"'.format(f))
```
Comment (takluyver) [inline tools/update_whatsnew.py]: This adds one bit of information - what file the error was on - but removes another: what error occurred.

Maybe print the filename and then reraise the original error?

File: tools/update_whatsnew.py (hunk 2 above)
Comment (Carreau) [inline tools/update_whatsnew.py]: I think I tried and this add the list to the set (got `List is unhashable` error).

I can try to unpack `set(*glob(pjoin(pr_dir, '*.rst')))`

File: docs/source/whatsnew/pr/do_where.rst [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    ----> 7     raise ValueError('6 is not acceptable')
+        8     7
+        9     8
+
+    ipdb> where 1
+    <ipython-input-3>(1)<module>()
+    ----> 1 foo()
+
+    > <ipython-input-2>(7)foo()
+    ----> 7     raise ValueError('6 is not acceptable')
```
Comment (takluyver) [inline docs/source/whatsnew/pr/do_where.rst]: I don't think we're going to want to illustrate this at such length in the whatsnew document. I know we can always trim it down later, but if you could write a shorter description now, that will make life easier when we do come to release time.

**Comment by takluyver** [comment]
Did the code change this is talking about already land? From the title I was expecting to see some diff in the debugger.

File: tools/update_whatsnew.py (hunk 2 above)
Comment (takluyver) [inline tools/update_whatsnew.py]: Not sure what you tried, but `set(glob(...))` works.

Comment (Carreau) [inline tools/update_whatsnew.py]: done.

File: tools/update_whatsnew.py (hunk 3 above)
Comment (Carreau) [inline tools/update_whatsnew.py]: done.

File: tools/update_whatsnew.py (hunk 4 above)
Comment (Carreau) [inline tools/update_whatsnew.py]: God point, that was my intention initially. I likely got distracted.

**Comment by Carreau** [comment]
> Did the code change this is talking about already land? From the title I was expecting to see some diff in the debugger.

Yes.  I added the number of the PR (#9097)

File: docs/source/whatsnew/pr/do_where.rst [hunk 6]
```diff
@@ -0,0 +1,3 @@
+IPython debugger (IPdb) now supports the number of context lines for the
+``where`` (and ``w``) commands. The `context` keyword is also available in various APIs.
+See PR #9097
```
Comment (minrk) [inline docs/source/whatsnew/pr/do_where.rst]: There's a `:ghpull:` roll that generates links to pull requests.

Comment (Carreau) [inline docs/source/whatsnew/pr/do_where.rst]: Done.

**Pull request merged**